### ISSUE TITLE
fast-eslint cannot to find the required module if `node_modules` isn't in the top-level directory.

### ISSUE THREAD

**Issue by twang2218** [Adezandee/fast-eslint#168]
To reproduce the problem, just create a subdirectory, and create a project with npm, and installed every eslint dependencies.

```bash
$ mkdir -p ~/a/b
$ npm init -y
$ npm i -D eslint \
        eslint-config-airbnb \
        eslint-plugin-import \
        eslint-plugin-jsx-a11y \
        eslint-plugin-react
$ touch index.js
$ touch .eslintrc
```

Then put some rules inside the `.eslintrc`, such as:

```json
{
  "extends": "airbnb",
  "rules": {
    "max-len": [1, 120, 2, {
      "ignoreComments": true
    }],
    "no-param-reassign": 0,
    "no-console": 0,
  }
}
```

Let's try run `atom` in the directory `~/a/b`, 

```bash
~/a/b $ atom .
```

Type something in `index.js` to verify the `eslint` is working.

Now, let's try run `atom` in the directory `~/a`

```bash
~/a/b $ cd ..
~/a $ atom .
```

Then type something in `index.js`, we got many errors about cannot find the modules.

```bash
Error: Cannot find module 'eslint-config-airbnb' Referenced from: /Users/taowang/a/b/.eslintrc
Error: Cannot find module 'eslint-config-airbnb'
Referenced from: /Users/taowang/a/b/.eslintrc
    at Object.ModuleResolver.resolve (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/util/module-resolver.js:75:19)
    at resolve (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config/config-file.js:479:33)
    at load (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config/config-file.js:496:24)
    at /Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config/config-file.js:392:36
    at Array.reduceRight (native)
    at applyExtends (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config/config-file.js:363:28)
    at Object.load (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config/config-file.js:530:22)
    at loadConfig (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config.js:64:33)
    at getLocalConfig (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config.js:136:23)
    at Config.getConfig (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/config.js:263:22)
    at processText (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/cli-engine.js:231:27)
    at CLIEngine.executeOnText (/Users/taowang/.atom/packages/fast-eslint/node_modules/eslint/lib/cli-engine.js:761:26)
    at /Users/taowang/.atom/packages/fast-eslint/lib/exec.js:73:24
    at exports.allowUnsafeNewFunction (/Users/taowang/.atom/packages/fast-eslint/node_modules/loophole/lib/loophole.js:25:14)
    at /Users/taowang/.atom/packages/fast-eslint/lib/exec.js:72:7
    at exec (/Users/taowang/.atom/packages/fast-eslint/lib/exec.js:69:10)
    at Object.lint (/Users/taowang/.atom/packages/fast-eslint/lib/index.js:48:16)
    at promises.push.Promise.then._this.emitter.emit.linter (/Users/taowang/.atom/packages/linter/lib/linter-registry.js:63:26)
    at /Users/taowang/.atom/packages/linter/lib/linter-registry.js:62:23
    at Set.forEach (native)
    at LinterRegistry.lint (/Users/taowang/.atom/packages/linter/lib/linter-registry.js:60:18)
    at /Users/taowang/.atom/packages/linter/lib/linter.coffee:98:16
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Users/taowang/.atom/packages/linter/lib/editor-linter.js:86:22
    at /Users/taowang/.atom/packages/linter/lib/helpers.coffee:25:18
```

**Comment by Adezandee** [comment]
Hello @twang2218 I'll give it a look when back from holidays, thanks for your report

**Comment by Adezandee** [comment]
Hello @twang2218,
I've tried to reproduce the errors, but I couldn't see it... here is a screenshot of my setup with some debugging logs.
![screen shot 2016-09-02 at 20 45 23](https://cloud.githubusercontent.com/assets/4415204/18214698/77f75020-714e-11e6-81de-93060c38ebbd.png)

**Comment by Adezandee** [comment]
Probably resolved by upgrading to eslint `3.5.0`, closing for now

**Comment by ishcristal** [comment]
hi @Adezandee ,

I can't help but notice that the node_modules being loaded is not the node_modules under /a/b. My directory looks like this:

```
+ a
+--+ b
+--+--+ node_modules
+--+--+ .eslintrc
+--+ c
+--+--+ node_modules
+--+--+ .eslintrc
```

it seems it's trying to load `/a/node_modules` when running eslint
